src/objects/flowRunArtifacts.ts

```diff
@@ -1,14 +1,20 @@
 import { flowRunArtifactsFactory } from '@/factories/flowRunArtifacts'
 import { RunGraphData } from '@/models/RunGraph'
 import { emitter } from '@/objects/events'
+import { waitForRunData } from '@/objects/nodes'
 
 export async function startFlowRunArtifacts(): Promise<void> {
+  const data = await waitForRunData()
```

Comment:
I learned the hard way that awaiting settings in flowRunArtifactsFactory was causing the render function to return after data was created/updated, so they wouldn't render initially. Now this object always checks for artifacts after data has been initialized.